### PR TITLE
vkd3d-shader: Fix modifier handling.

### PR DESCRIPTION
**Pull request by doitsujin**
This is a set of flags, so we should treat it as such.

### CODE REVIEW

File: libs/vkd3d-shader/dxbc.c [hunk 1]
```diff
@@ -1538,30 +1538,19 @@ static bool shader_sm4_read_param(struct vkd3d_sm4_data *priv, const DWORD **ptr
...
+        switch (m & 0xc0)
         {
-            case 0x41:
-                *modifier = VKD3DSPSM_NEG;
-                break;
+            case 0x00: *modifier = VKD3DSPSM_NONE; break;
+            case 0x40: *modifier = VKD3DSPSM_NEG; break;
+            case 0x81: *modifier = VKD3DSPSM_ABS; break;
+            case 0xc1: *modifier = VKD3DSPSM_ABSNEG; break;
+        };
```
Comment (HansKristian-Work) [inline libs/vkd3d-shader/dxbc.c:1547]: weird semi-colon

File: libs/vkd3d-shader/dxbc.c [hunk 2]
```diff
@@ -1538,30 +1538,19 @@ static bool shader_sm4_read_param(struct vkd3d_sm4_data *priv, const DWORD **ptr
...
 
-        switch (m)
+        switch (m & 0xc0)
         {
-            case 0x41:
-                *modifier = VKD3DSPSM_NEG;
-                break;
+            case 0x00: *modifier = VKD3DSPSM_NONE; break;
+            case 0x40: *modifier = VKD3DSPSM_NEG; break;
+            case 0x81: *modifier = VKD3DSPSM_ABS; break;
```
Comment (HansKristian-Work) [inline libs/vkd3d-shader/dxbc.c:1545]: Can 0x81 or 0xc1 occur here?

Comment (doitsujin) [reply libs/vkd3d-shader/dxbc.c:1545]: Typo, this should be 0x80. 1 at the end can never occur because it's masked out.

Comment (doitsujin) [reply libs/vkd3d-shader/dxbc.c:1545]: Ideally we'd have the flags defined somewhere, but the headers currently don't do that...

**Pull request merged**